### ISSUE TITLE
ShellJob: convert `argument` when it is a string.

### ISSUE THREAD

**Issue by superstar54** [aiidateam/aiida-workgraph#315]
The current implementation of [prepare_for_shell_task](https://github.com/aiidateam/aiida-workgraph/blob/5bf734af360b12e238ac8955f8eb0512fb19fb29/aiida_workgraph/engine/utils.py#L132-L164) lacks the functionality to handle the arguments parameter when it is provided as a string, which is present in the [launch_shell_job](https://github.com/sphuber/aiida-shell/blob/104d03b3b9b43b3b3eec77e6e10bd6ec7ffa5ec0/src/aiida_shell/launch.py#L82-L85) function.

**Comment by GeigerJ2** [comment]
Just a note for myself while working on this: If I have a simple WG with one `ShellJob` task, the `prepare_for_shell_task` function is being called when I execute `wg.run()` but not when I do `wg.submit()`, e.g.:

```python
arguments = '--iso-8601 --universal'
wg = WorkGraph(name="test_shell_date")
date_task = wg.add_task("ShellJob", command="date", arguments=arguments)
wg.run()  # Here, changes in `prepare_for_shell_task` are reflected
```

vs.:

```python
arguments = '--iso-8601 --universal'
wg = WorkGraph(name="test_shell_date")
date_task = wg.add_task("ShellJob", command="date", arguments=arguments)
wg.submit()  # Here, changes in `prepare_for_shell_task` are **not** reflected
```

**Comment by superstar54** [comment]
Did you restart the daemon?

**Comment by GeigerJ2** [comment]
Always the same fkn issue 😆 yeah, that was probably it...

**Comment by superstar54** [comment]
closed by #326
